htlcswitch/hop/payload.go

```diff
@@ -258,15 +260,16 @@ func NewCustomRecords(parsedTypes tlv.TypeMap) record.CustomSet {
 // ensure that the proper fields are either included or omitted. The finalHop
 // boolean should be true if the payload was parsed for an exit hop. The
 // requirements for this method are described in BOLT 04.
-func ValidateParsedPayloadTypes(parsedTypes tlv.TypeMap,
-	isFinalHop bool) error {
+func ValidateParsedPayloadTypes(parsedTypes tlv.TypeMap, isFinalHop,
```

Comment:
re the feature bit comment made in the next PR, perhaps that could be checked in one go here?

Reply:
The spec doesn't _say_ that we should fail if an unadvertised feature is used, will double check whether there's any instructions for this sort of case! But let's keep the main discussion on the [next PR ](https://github.com/lightningnetwork/lnd/pull/8485#discussion_r1505581314)so we don't fragment it across two.